### PR TITLE
Make Cost and CostMatrix abstract classses

### PR DESCRIPTION
**Pull request by bartvm**

### CODE REVIEW

File: blocks/bricks/cost.py [hunk 1]
```diff
@@ -18,6 +26,11 @@ def apply(self, y, y_hat):
         return self.cost_matrix.application_method(
             self, y, y_hat).sum(axis=1).mean()
 
+    @abstractmethod
+    @application
```
Comment (rizar) [inline blocks/bricks/cost.py]: Did you try that such nested decorators work?

**Comment by rizar** [comment]
Nice, merge when you want. I should say that `cost.py` definitely needs more docs...

File: blocks/bricks/cost.py (hunk 1 above)
Comment (bartvm) [inline blocks/bricks/cost.py]: Should do, I've used it before. It's just he order that becomes important
usually, but in this case they're complementary, so it should be good.

**Comment by bartvm** [comment]
Agreed!

**Pull request merged**